apps/nextjs/src/app/api/auth/[...nextauth]/route.ts

```diff
@@ -1,18 +1,42 @@
-import type { NextRequest } from "next/server";
 import { cookies } from "next/headers";
-import { NextResponse } from "next/server";
+import { NextRequest, NextResponse } from "next/server";
 
-import { GET as DEFAULT_GET, POST } from "@acme/auth";
+import { GET as DEFAULT_GET, POST as DEFAULT_POST } from "@acme/auth";
+
+import { env } from "~/env";
 
 export const runtime = "edge";
 
 const EXPO_COOKIE_NAME = "__acme-expo-redirect-state";
 const AUTH_COOKIE_PATTERN = /authjs\.session-token=([^;]+)/;
 
+/**
+ * Correct request.url for local development so that Expo can work. Does nothing in production.
+ * @param req The request to modify
+ * @returns The modified request.
+ */
+function rewriteRequestUrl(req: NextRequest) {
+  if (env.NODE_ENV === 'production') {
+    return req;
+  }
+  const host = req.headers.get("host");
+  const newURL = new URL(req.url);
+  newURL.host = host ?? req.nextUrl.host;
+  return new NextRequest(newURL, req);
```

Comment:
Why do we need to rewrite if we disable csrf in dev?

Reply:
Because the generated callback URLs point to `localhost:3000` no matter what. NextJS sets the `nextUrl` to the deployment URL, which is always `localhost` on development, but we need it to be the IP of the machine, i.e. `192.168.x.y` so that it can properly route the request to the oauth provider and back.

Reply:
Having looked through authjs's source code, the callback url is generated by `const baseUrl = env.AUTH_URL ?? request.url` (pseudocode), so because we don't set `AUTH_URL`, we need to ensure `request.url` is set to the proper value.

Reply:
So this means we have to set the ip at the oauth provider? Hmm...

Reply:
Just modifying the request URL seems to be sufficient. In my local testing, this has worked great both with and without the auth proxy server.

Reply:
Aight I'll try and test it out asap